src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Automation/VsBuildManager.cs

```diff
@@ -0,0 +1,180 @@
+        /// </summary>
+        public event _dispBuildManagerEvents_DesignTimeOutputDeletedEventHandler DesignTimeOutputDeleted;
+
+        /// <summary>
+        /// Undocumented.
+        /// </summary>
+        public event _dispBuildManagerEvents_DesignTimeOutputDirtyEventHandler DesignTimeOutputDirty;
+
+        #endregion
+
+        /// <summary>
+        /// Undocumented.
+        /// </summary>
+        public virtual EnvDTE.Project ContainingProject
+        {
+            get { return _vsProject.Project; }
+        }
+
+        /// <summary>
+        /// Undocumented.
+        /// </summary>
+        public virtual EnvDTE.DTE DTE
+        {
+            get { return _vsProject.DTE; }
+        }
+
+        /// <summary>
+        /// Undocumented.
+        /// </summary>
+        public virtual object DesignTimeOutputMonikers
+        {
+            get
+            {
+                var project = _unconfiguredProjectServices.ActiveConfiguredProject;
+                var ruleSource = project.Services.ProjectSubscription.ProjectRuleSource;
+                var update = _unconfiguredProjectServices.ThreadingService.ExecuteSynchronously(() => ruleSource.GetLatestVersionAsync(project, new string[] { Compile.SchemaName }));
+                var snapshot = update[Compile.SchemaName];
+
+                var monikers = new List<string>();
+                foreach (var item in snapshot.Items.Values)
+                {
+                    bool isLink = GetBooleanPropertyValue(item, Compile.LinkProperty);
+                    bool designTime = GetBooleanPropertyValue(item, Compile.DesignTimeProperty);
+
+                    if (!isLink && designTime)
+                    {
+                        if (item.TryGetValue(Compile.FullPathProperty, out string path))
+                        {
+                            monikers.Add(path);
+                        }
+                    }
+                }
+                
+                return monikers.ToArray();
+
+                bool GetBooleanPropertyValue(System.Collections.Immutable.IImmutableDictionary<string, string> item, string propertyName)
+                {
+                    return item.TryGetValue(propertyName, out string value) && StringComparers.PropertyValues.Equals(value, "true");
+                }
+            }
+        }
+
+        /// <summary>
+        /// Undocumented.
+        /// </summary>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1065:DoNotRaiseExceptionsInUnexpectedLocations")]
+        public virtual object Parent
+        {
+            get { throw new NotImplementedException(); }
+        }
+
+        #region BuildManager Members
+
+        /// <summary>
+        /// Undocumented.
+        /// </summary>
+        public virtual string BuildDesignTimeOutput(string bstrOutputMoniker)
+        {
+            var languageName = _unconfiguredProjectServices.ThreadingService.ExecuteSynchronously(async () =>
```

Comment:
Validate the bstroutputMoniker the same way legacy does.

Reply:
Why is this virtual?

Reply:
LanguageName can be empty here.

Reply:
Any method that uses Roslyn service should be switching to the UI thread, basically should wrap this method in ExecuteSync.